src/app/workflow/view/view.component.ts

```diff
@@ -89,6 +93,35 @@ export class ViewWorkflowComponent extends View implements OnInit {
     });
   }
 
+  private updateWorkflowToSnapshot(version): void {
+    this.workflowsService.updateWorkflowVersion(this.workflow.id, [version]).subscribe(
+      workflowVersions => {
+        this.alertService.detailedSuccess('Snapshot successfully created!');
+        this.refreshService.refreshWorkflow();
+      },
+      error => {
+        if (error) {
+          this.alertService.detailedError(error);
+        } else {
+          this.alertService.simpleError();
+        }
+      }
+    );
+  }
+
+  snapshotVersion(): void {
+    // Create a new version temporarily with the frozen bit set. We assume the version is
+    // not already a snapshot since the UI controls shouldn't be available for
+    // snapshotted versions.
```

Comment:
I can tell you come from Python :smile:

Reply:
lol I can think of a few reason. 1) I am trying to comment our way around a mutation operation. I presume there is a way to use the app data model to allow safe mutation operations without needing to create temporary variables? 2) My comments are not blocked. 

In either case let me know! I know that years of Python can have its toll on a person.

Reply:
I initially thought the comments were for the method.  Python usually has the comments below the method name, but Typescript usually has it above.